jetty-server/src/main/java/org/eclipse/jetty/server/session/HouseKeeper.java

```diff
@@ -158,7 +158,7 @@ protected void stopScavenging() throws Exception
                 LOG.info("{} Stopped scavenging", _sessionIdManager.getWorkerName());
             }
             _task = null;
-            if (_ownScheduler)
+            if (_ownScheduler && _scheduler != null)
```

Comment:
I think the fix is to set _ownScheduler = false here after line 164. That way _ownScheduler and _scheduler are kept in sync.

Reply:
Thank you for your review. I agree and fixed it.